### PR TITLE
Promote + Demote Functionality for Admin Dashboard

### PR DESCRIPTION
**Pull request by mishasavlv**
# Pull Request

## Description

Fixes promote buggy functionality. In the admin dashboard for the promote modal, I added where when the Promote button is clicked for a Participant the live list of university options from firestore is pulled and sent to the Promote modal as a prop, where the Promote modal displays the options in the drop down. When the admin submits a school, the choice is validated before submit to ensure it is still an option in Firestore, covering the edge case where a school is removed while the admin has the modal open. When a participant is promoted, in Firestore their role is changed to 'Subadmin' and the selected university is added under a field with name 'university'.

Demote button and modal were also added in admin dashboard next to subadmin users. It will demote them back to participant, changing their role in Firestore back to 'Participant' and removing the university field.

## Type of Change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Refactoring (no functional changes)
- [ ] Performance improvement
- [ ] Other (please describe):

## Related Issues (put task name here from Notion)

Cleanup “Promote” feature and add support for demotion - https://app.notion.com/p/h4i/Cleanup-Promote-feature-and-add-support-for-demotion-383197abf07b80e39a32c5adbcae34e8?source=copy_link

### Code Review Preparation
- [x] I have performed a self-review and reviewed the changed files
- [x] My code follows the project's style guidelines and passes linting
- [x] I have added comments in hard-to-understand areas and updated documentation

### Testing
- [x] I have added relevant tests and they pass locally
- [x] I have tested the changes manually, including edge cases

### Code Quality & Security
- [x] I have formatted my code and removed debugging/unused code
- [x] No TypeScript errors or linting issues
- [x] No sensitive information committed (API keys, passwords, etc.)
- [x] User inputs are validated and sanitized

## Screenshots/Screen recording (If applicable)

[Added screen recording of each feature.](https://github.com/user-attachments/assets/2f82e2ae-f3e6-455d-95a6-5916ae58fdcb)

## Summary by CodeRabbit

* **New Features**
  * Added support to demote admin subaccounts back to participant status, with a dedicated demotion confirmation dialog.
  * Updated role-aware admin controls so the user management table shows Promote/Demote actions based on eligibility.
* **Improvements**
  * Strengthened the admin promotion flow with live university availability checks to prevent unsupported selections.
  * Refined promotion/demotion modal behavior for smoother transitions and clearer success/error messaging.
  * Improved phone number formatting and waitlist badge styling.

### CODE REVIEW

File: src/pages/Users/AdminCreator.tsx [hunk 1]
```diff
@@ -591,12 +664,42 @@ function PromoteModal({ participant, onClose, onSuccess }: PromoteModalProps) {
...
+    // edge case handling - after submitting request to promote or demote, get the
+    // live list of universities from firestore to check that the selected one is still on the list
+    const liveUniversityOptions = await getUniversityOptions();
+    if (!liveUniversityOptions.includes(university)) {
+      if (liveUniversityOptions.length > 0) {
+        setError(
+          "The college you selected is no longer supported. Please select a new option.",
+        );
+      } else {
+        alert("The college you selected is no longer supported.");
```
Comment (evtran4) [inline src/pages/Users/AdminCreator.tsx]: Can this be combined with the above error instead of showing as a browser alert? We generally try to avoid using those.

File: src/pages/Users/AdminCreator.tsx [hunk 2]
```diff
@@ -245,6 +259,18 @@ export default function AdminDashboard() {
       });
   }, [selectedUser]);
 
+  // useEffect to trigger the promote modal when the
+  // promoteState.adminForModal state var is not null
+  useEffect(() => {
```
Comment (evtran4) [inline src/pages/Users/AdminCreator.tsx]: Can this be simplified at all to stop using this useEffect and adminForModal? What if we instead just used an async click handler kind of like this:


```
onClick={async () => {
  const universityOptions = await getUniversityOptions();
  setPromoteTarget(admin);
  setPromoteState((prev) => ({
    ...prev,
    universities: universityOptions,
  }));
}}
```

File: src/pages/Users/AdminCreator.tsx [hunk 3]
```diff
@@ -655,8 +758,23 @@ function PromoteModal({ participant, onClose, onSuccess }: PromoteModalProps) {
               autoFocus
             />
           </div> */}
-          <DropdownInput name={"Uni"} options={[]} required={false}>
-          </DropdownInput>
+          {/* <DropdownInput
```
Comment (evtran4) [inline src/pages/Users/AdminCreator.tsx]: Commented out code can be removed

**Review by evtran4 — COMMENTED** [review]
Forgot to hit submit my bad

File: src/pages/Users/AdminCreator.tsx (hunk 1 above)
Comment (mishasavlv) [reply src/pages/Users/AdminCreator.tsx]: addressed in commit b5367b7

File: src/pages/Users/AdminCreator.tsx (hunk 3 above)
Comment (mishasavlv) [reply src/pages/Users/AdminCreator.tsx]: addressed in commit a3229d2

File: src/pages/Users/AdminCreator.tsx (hunk 2 above)
Comment (mishasavlv) [reply src/pages/Users/AdminCreator.tsx]: addressed in commit 12aa35a


initially I chose to do it the way I did originally with the useEffect because I thought that the two state changes would happen asynchronously and the promote modal wouldn't be able to see the university list. But I realize my understanding was flawed.

**Review by evtran4 — APPROVED** [review]
